src/shared/dissect-image.c

```diff
@@ -1266,6 +1287,12 @@ DissectedImage* dissected_image_unref(DissectedImage *m) {
          * DecryptedImage may try to deactivate partitions. */
         decrypted_image_unref(m->decrypted_image);
 
+        if (do_forget) {
+                r = btrfs_forget_device(m->loop->fd);
```

Comment:
is this right? this is an fd to the main loopback device, not the partition block device the btrfs file system will be on, no?

Reply:
I verified it fixes the issue, so it should be good

Reply:
so you are saying btrfs automatically iterates through all child devices too? wild

Reply:
Nvm, turns out the `fd` arg is just ignored by this ioctl(), I'll have to rework to pass the loop device path, let's see if it still works then.